### PR TITLE
SAAS-491 registration with okta

### PR DESCRIPTION
**Pull request by nicolalamacchia**

### CODE REVIEW

File: packages/platform-ui/src/core/api/auth.ts [hunk 1]
```diff
@@ -26,13 +26,15 @@ export const signIn: SignIn = ({ email, password }) => {
   return apiClient.signIn(request, {});
 };
 
-export const signUp: SignUp = ({ email, password }) => {
+export const signUp: SignUp = ({ email, firstName = 'First', lastName = 'Last' }) => {
```
Comment (tiagomotasantos) [inline packages/platform-ui/src/core/api/auth.ts:29]: Would put these default strings the messages file too.

File: packages/platform-ui/src/store/auth/auth.reducer.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export const authSignupAction = createAsyncAction(
   'SIGNUP_USER_REQUEST',
   'SIGNUP_USER_SUCCESS',
   'SIGNUP_USER_FAILURE',
-)<{ email: string; password: string }, undefined, grpcWeb.Error>();
+)<{ email: string; firstName: string, lastName: string, password: string }, undefined, grpcWeb.Error>();
```
Comment (tiagomotasantos) [inline packages/platform-ui/src/store/auth/auth.reducer.ts:28]: Better to use an interface here since this is getting big.

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]

File: packages/platform-ui/src/core/api/auth.ts (hunk 1 above)
Comment (atymchuk) [inline packages/platform-ui/src/core/api/auth.ts:29]: Are those defaults? I don't see why we would put these values as defaults... If no firstName is provide, the API should just fail.

File: packages/platform-ui/src/core/api/messages.ts [hunk 3]
```diff
@@ -5,5 +5,5 @@ export const Messages = {
   signOutFailed: 'Logout failed, please try again',
   signOutSucceeded: 'You are now logged out',
   signUpFailed: 'There was a problem during the sign up process, please try again',
-  signUpSucceeded: 'You have successfully created your credentials',
+  signUpSucceeded: 'A confirmation email has been sent to you',
```
Comment (atymchuk) [inline packages/platform-ui/src/core/api/messages.ts:8]:
```suggestion
  signUpSucceeded: 'An account activation email has been sent to you',
```

File: packages/platform-ui/src/core/gen/web/auth/auth_api_pb.js [hunk 4]
```diff
@@ -267,7 +267,9 @@ proto.percona.platform.auth.v1.SignUpRequest.prototype.toObject = function(opt_i
 proto.percona.platform.auth.v1.SignUpRequest.toObject = function(includeInstance, msg) {
   var f, obj = {
     email: jspb.Message.getFieldWithDefault(msg, 1, ""),
-    password: jspb.Message.getFieldWithDefault(msg, 2, "")
+    password: jspb.Message.getFieldWithDefault(msg, 2, ""),
```
Comment (atymchuk) [inline packages/platform-ui/src/core/gen/web/auth/auth_api_pb.js:270]: Do we no longer generate the code?

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: packages/platform-ui/src/core/gen/web/auth/auth_api_pb.js (hunk 4 above)
Comment (nicolalamacchia) [reply packages/platform-ui/src/core/gen/web/auth/auth_api_pb.js:270]: We still rely on generated code, we're going to change this soon though

**Review by fabio-silva — DISMISSED** [review]

File: packages/platform-ui/src/store/auth/auth.reducer.ts (hunk 2 above)
Comment (nicolalamacchia) [reply packages/platform-ui/src/store/auth/auth.reducer.ts]: Done

**Review by fabio-silva — DISMISSED** [review]

**Review by tiagomotasantos — COMMENTED** [review]
Some auth related tests are failing, other then that LGTM.

File: packages/platform-ui/cypress/integration/auth/signup.spec.ts [hunk 5]
```diff
@@ -1,47 +1,88 @@
...
+    cy.task('setPassword', newUser.user.password);
+
+    cy.visit(pageDetailsMap[Pages.SignUp].url);
     runSignUpFlow(newUser);
-    runLoginAction(newUser.user);
-    userEmail().isVisible().hasText(newUser.user.email);
-    profileIcon().isVisible();
+  });
+
+  it('should not be able to login with not activated account', () => {
```
Comment (fabio-silva) [inline packages/platform-ui/cypress/integration/auth/signup.spec.ts:45]: "not activated account" -> "inactive account"?

File: packages/platform-ui/cypress/integration/auth/signup.spec.ts [hunk 6]
```diff
@@ -1,47 +1,88 @@
...
+      fillEmailPassword(newUser.user);
+      submitButton().isVisible().isEnabled().click();
+      popUp().isVisible().hasText(INVALID_USER.invalidLoginMessage);
+      loginForm().isVisible();
+    });
+  });
+
+  it('should be able to activate account', { baseUrl: null}, () => {
+    cy.task('getUser').then(({ userEmail, userPassword }) => {
+      cy.mailosaurGetMessage(Cypress.env('MAILOSAUR_SAAS_SERVER_ID'), {sentTo: userEmail}, {timeout: 20000})
```
Comment (fabio-silva) [inline packages/platform-ui/cypress/integration/auth/signup.spec.ts:59]: some spacing missing before/after brackets

File: packages/platform-ui/cypress/pages/auth/flows/validation.flow.ts [hunk 7]
```diff
@@ -10,31 +13,46 @@ import {
...
   submitButton().isVisible().isDisabled();
 
   if (page === Pages.SignUp) {
-    fillField({ field: emailField, value: VALID_USER.user.email });
+    fillField({ field: emailField, value: EXISTING_USER.user.email });
+    firstNameField().focus();
+    lastNameField().focus();
+    firstNameField().focus();
+    checkValidation({element: firstNameValidation, text: VALIDATION_MESSAGES.REQUIRED_FIELD});
+    checkValidation({element: lastNameValidation, text: VALIDATION_MESSAGES.REQUIRED_FIELD});
```
Comment (fabio-silva) [inline packages/platform-ui/cypress/pages/auth/flows/validation.flow.ts:47]: spacing problem as well

File: packages/platform-ui/cypress/pages/auth/utils/getUser.ts [hunk 8]
```diff
@@ -21,17 +21,18 @@ export const getPassword = () => {
...
     user: {
       email,
       password,
+      firstName: faker.name.firstName(),
+      lastName: faker.name.lastName(),
     },
-    signedInMessage: `You are signed in as ${email}`,
-    signedUpMessage: 'You have successfully created your credentials',
+    signedInMessage,
+    activationEmailSentMessage: 'An account activation email has been sent to you',
```
Comment (fabio-silva) [inline packages/platform-ui/cypress/pages/auth/utils/getUser.ts:35]: aren't we repeating the same messages defined in `packages/platform-ui/src/core/api/messages.ts` ?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: packages/platform-ui/cypress/integration/auth/signup.spec.ts (hunk 6 above)
Comment (nicolalamacchia) [reply packages/platform-ui/cypress/integration/auth/signup.spec.ts:59]: @yurkovychv this is a nit, but your linter might need some adjustments ;P

File: packages/platform-ui/cypress/pages/auth/utils/getUser.ts (hunk 8 above)
Comment (nicolalamacchia) [reply packages/platform-ui/cypress/pages/auth/utils/getUser.ts:35]: Those are issues (nitpicks) that were already there or are relevant to e2e. For the sake of speed, let’s leave them for now :)

Comment (yurkovychv) [reply packages/platform-ui/cypress/pages/auth/utils/getUser.ts:35]: @nicolalamacchia @fabio-silva Agree, it was done in rush to not block FE merges. I plan to refactor those tests in a near time

**Review by fabio-silva — APPROVED** [review]

**Review by tiagomotasantos — APPROVED** [review]

File: packages/platform-ui/cypress/pages/auth/flows/validation.flow.ts [hunk 9]
```diff
@@ -2,6 +2,9 @@ import {
   emailField,
   emailFieldLabel,
   emailValidation,
+  firstNameField,
+  firstNameValidation,
+  lastNameField, lastNameValidation,
```
Comment (atymchuk) [inline packages/platform-ui/cypress/pages/auth/flows/validation.flow.ts:7]: nit: need to break it to two lines tbc

**Review by atymchuk — APPROVED** [review]

**Pull request merged**